kotsadm/web/src/components/modals/UpdateCheckerModal.jsx

```diff
@@ -0,0 +1,113 @@
+import React from "react";
+import Modal from "react-modal";
+import { Utilities, getReadableCronDescriptor } from "../../utilities/utilities";
+
+export default class UpdateCheckerModal extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      updateCheckerSpec: props.updateCheckerSpec,
+      submitUpdateCheckerSpecErr: ""
+    };
+  }
+
+  onSubmitUpdateCheckerSpec = () => {
+    const { updateCheckerSpec } = this.state;
+    const { appSlug } = this.props;
+
+    this.setState({
+      submitUpdateCheckerSpecErr: ""
+    });
+
+    fetch(`${window.env.API_ENDPOINT}/app/${appSlug}/updatecheckerspec`, {
+      headers: {
+        "Authorization": Utilities.getToken(),
+        "Content-Type": "application/json",
+      },
+      method: "PUT",
+      body: JSON.stringify({
+        updateCheckerSpec: updateCheckerSpec,
+      })
+    })
+      .then(async (res) => {
+        if (!res.ok) {
```

Comment:
The handler always returns a JSON object on error, with the message in `error` key.  This does not use that message.  This will display a JSON string to the user.